pkg/build/generator/generator.go

```diff
@@ -575,25 +575,64 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from kapi
 	glog.V(4).Infof("Resolving ImageStreamReference %s of Kind %s in namespace %s", from.Name, from.Kind, namespace)
 	switch from.Kind {
 	case "ImageStreamImage":
-		imageStreamImage, err := g.Client.GetImageStreamImage(kapi.WithNamespace(ctx, namespace), from.Name)
+		name, id, err := imageapi.ParseImageStreamImageName(from.Name)
 		if err != nil {
 			err = resolveError(from.Kind, namespace, from.Name, err)
 			glog.V(2).Info(err)
 			return "", err
 		}
-		image := imageStreamImage.Image
-		glog.V(4).Infof("Resolved ImageStreamReference %s to image %s with reference %s in namespace %s", from.Name, image.Name, image.DockerImageReference, namespace)
-		return image.DockerImageReference, nil
+		stream, err := g.Client.GetImageStream(kapi.WithNamespace(ctx, namespace), name)
+		if err != nil {
+			err = resolveError(from.Kind, namespace, from.Name, err)
+			glog.V(2).Info(err)
+			return "", err
+		}
+		tag := ""
+		for tagName, imageList := range stream.Status.Tags {
```

Comment:
This will pick up tags in random order, which means I can have different results for two runs on exact same ImageStream.

If I have an image tagged under `is:tag` somewhere in an ancient history and I re-tag it again as `is:latest` and I update the tag with:

```
oc patch istag/is:latest -p '{"tag":{"referencePolicy":{"type":"Local"}}}'
```

I'd probably expect that all the new build(configs) will resolve it as a local image.

With this code, I get completely random result.